### ISSUE TITLE
Document and test the scrambling of the original data for all 5 use cases

### ISSUE THREAD

**Issue by robertoostenveld** [SIESTA-eu/wp15#46]
@schoffelen encountered an empty json file in #31, which lead to errors when reading the EEG data with FieldTrip (as it is not BIDS valid with an empty json). This reminded me that we should test and review the scrambling of all 5 use cases. 

The goal is that all scrambled datasets should be as valid (*) as the input datasets. 

(*) Note that I write "as valid" and not "perfectly valid", since for some BIDS datasets (like 2.1 and 2.3) we only download a subsection which would make the dataset incomplete and hence not pass the BIDS validator to start with.

**Comment by marcelzwiers** [comment]
Sure, but then it's the missing fields that are not BIDS-compliant. Using the inheritance principle, it can be just fine to have empty jsons in subfolders

**Comment by robertoostenveld** [comment]
I am not sure whether the json file can be empty, or whether they should be absent in case of inheritance; the examples that I have seen so far (like [this](https://github.com/bids-standard/bids-examples/tree/master/eeg_matchingpennies)) omit the files down in the tree. But according to 5b in https://bids-specification.readthedocs.io/en/stable/common-principles.html#rules it appears that empty files (*) might be allowed when all required metadata fields are specified at a higher level. 

*) However, https://jsonlint.com is of the opinion that an empty string is not valid json. That aligns with https://www.json.org/json-en.html, according to which valid JSON needs at least some characters, like `{}`, `[]` or `""`. 

The fact remains that Jan-Mathijs encountered a dataset that was invalid due to empty json files which obviously were missing the required fields, and these required fields were also not available in a metadata json file at a higher level. Hence the scrambling instructions for the example datasets must be improved.

**Comment by marcelzwiers** [comment]
Yes, and empty string **value**, sure. But if there are no keys, then there is no empty string value either. I agree that empty jsons are not desirable, but they are not wrong either

**Comment by robertoostenveld** [comment]
The [JSON specification](https://www.json.org/json-en.html) is clear: a file with zero bytes is not a valid JSON file and `JSON.parse()` is to fail. It should have at least an array (aka list), an object, a string, a number, true or false, or null. An empty array `[]`, object `{}` or string `""` does make it a valid file. You can try it yourself on https://jsonlint.com.

Note that keys are only required in JSON objects, not in arrays, nor in strings.

**Comment by marcelzwiers** [comment]
Okay, I'll add a check in the `json` action for this, but I think `stub` is a generic action and should remain as is, i.e. write out empty files irrespective of their contents or format.

**Comment by robertoostenveld** [comment]
> However, the Siesta (test/dev) workflow is not explicitly described.

There is https://github.com/SIESTA-eu/wp15/blob/main/docs/README.md#data-flow

**Comment by robertoostenveld** [comment]
Regarding "1. data rights holder -> uploads input data to the platform", there is also the alternative that the platform owner downloads the data (i.e., one of the sides initiates the transfer, in one case we call it "upload", in the other "download"). @hansjurgen-wolf is now testing that as "platform owner" with #45.
